include/geom/elem.h

```diff
@@ -1080,6 +1080,12 @@ class Elem : public ReferenceCountedObject<Elem>,
    */
   bool positive_face_orientation(const unsigned int i) const;
 
+  /**
+   * \returns \p true if the vertices of edge \p e appear in edge_nodes_map in
+   * the same order they appear in side_nodes_map for side \p s.
```

Comment:
Pretend `edge_nodes_map` and `side_nodes_map` don't exist.  (I'm currently writing element classes where they don't.)  Rewrite with that in mind?

Something like "returns true iff the local node numbering of edge e matches the local node numbering of that edge when viewed as a side of side s"?

I'm also struggling a bit to understand the use of this.  Presumably you need a fallback code path for where it returns false?  Is it that much of an optimization to test it and use a faster code path when it returns true?  But it can't be, if you use it on every element, and yet not making it a `static` method suggests it might be true or false on an element-by-element basis.

I'd also hope to see some unit testing for this, perhaps something that demonstrates why it would be useful.

Reply:
> I'm also struggling a bit to understand the use of this.
> perhaps something that demonstrates why it would be useful.

Yep, my bad. It was always a risk to put this up without proper motivation.
In short, I'm trying to remove commits out the preconditioners PR-to-be (it's probably time I put this up, because I need your help with some design choices anyway).
For the ADS preconditioner, I need to know the relative orientation of faces to their defining edges and so I need to assess both orientations as if I'm cycling (clockwise for instance) through the face, one vertex after the other. That's the reason for this functionality (it works).

Reply:
Okay, I've removed that commit.